src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBaseSourceConfig.java

```diff
@@ -168,13 +169,14 @@ public String getFQN(String orgId, String sObject) {
     return String.format("salesforce://%s/%s.%s", firstFQNPart, orgId, sObject);
   }
 
-  public String getOrgId() throws ConnectionException {
-    PartnerConnection partnerConnection = SalesforceConnectionUtil.getPartnerConnection
-      (connection.getAuthenticatorCredentials());
+  public String getOrgId(OAuthInfo oAuthInfo) throws ConnectionException {
+    AuthenticatorCredentials credentials = new AuthenticatorCredentials(oAuthInfo,
+            this.getConnection().getConnectTimeout());
+    PartnerConnection partnerConnection = SalesforceConnectionUtil.getPartnerConnection(credentials);
     return partnerConnection.getUserInfo().getOrganizationId();
   }
 
-  protected void validateFilters(FailureCollector collector) {
+  public void validateFilters(FailureCollector collector) {
```

Comment:
Why is this made public?

Reply:
Previously it was getting called from its sub classes but now that we have segregated the code with connection to salesforce and without connection to salesforce. We have to call the validation methods that don't require OAuthInfo.